src/hermite/cubic.jl

```diff
@@ -0,0 +1,163 @@
+struct CubicHermite
+    xs::Vector{Float64}
+    ys::Vector{Float64}
+    dydxs::Vector{Float64}
+    function CubicHermite(xs, ys, dydxs)
+        if length(xs) < 2
+            throw(
+                DomainError(
+                    "There must be at least two samples in the domain of the CubicHermite interpolator.",
+                ),
+            )
+        end
+        x = xs[1]
+        for i = 2:length(xs)
+            if xs[i] <= x
+                throw(DomainError("The list of abscissas must be strictly increasing."))
+            end
+            x = xs[i]
+        end
+        if (length(xs) != length(ys) || length(ys) != length(dydxs))
+            throw(
+                DomainError(
+                    "There must be exactly the same number of abscissasas ordinates and derivatives.",
+                ),
+            )
+        end
+        new(xs, ys, dydxs)
+    end
+end
+
+function (ch::CubicHermite)(x::Float64)::Float64
+    if x <= ch.xs[1]
+        if x == ch.xs[1]
+            return ch.ys[1]
+        end
+        throw(
+            DomainError(
+                "Requested abscissas $x is less than the minimum value of the domain of the interpolator.",
+            ),
+        )
+    end
+
+    if x >= last(ch.xs)
+        if x == last(ch.xs)
+            return last(ch.ys)
+        end
+        throw(
+            DomainError(
+                "Requested abscissa $x is greater than the maximum value of the domain of the interpolator.",
+            ),
+        )
+    end
+    # searchsorted is convenient but it is a little too general for this usecase.
+    # We have asserted that the xs's are strictly increasing.
+    # Ostensibly there is another function that could do this more cleanly.
+    urange = searchsorted(ch.xs, x)
+    # urange.start exceeds urange.stop in this case which again indicates that searchsorted is not quite the right tool for this job:
+    i = urange.stop
+    x1 = ch.xs[i]
+    x2 = ch.xs[i+1]
+    @assert x1 <= x <= x2
+    h = x2 - x1
+    y1 = ch.ys[i]
+    y2 = ch.ys[i+1]
+
+    dydx1 = ch.dydxs[i]
+    dydx2 = ch.dydxs[i+1]
+    # Corless, A Graduate Introduction to Numerical Methods, equation 14.10:
+    c1 = (2x - 3x1 + x2) * (x - x2) * (x - x2) / (h * h * h)
+    c2 = -(2x - 3x2 + x1) * (x - x1) * (x - x1) / (h * h * h)
+    d1 = (x - x1) * (x - x2) * (x - x2) / (h * h)
+    d2 = (x - x2) * (x - x1) * (x - x1) / (h * h)
+    c1 * y1 + c2 * y2 + d1 * dydx1 + d2 * dydx2
```

Comment:
Codecov is saying the tests do not make it here?

Reply:
Should be fixed with the latest push.